fe1-web/src/features/evoting/reducer/ElectionKeyWatcher.ts

```diff
@@ -0,0 +1,20 @@
+import { Store } from 'redux';
+
+import { PublicKey } from 'core/objects';
+
+import { getElectionKeyByElectionId } from './ElectionKeyReducer';
+
+/**
+ * Watches the redux store for new election#key message for the given election id
+ * @remark Implemented analogous to 'makeMessageStoreWatcher'
+ * @param store The redux store to watch
+ * @param callback The function to call when a the corresponding election#key message is received
+ */
+export const makeElectionKeyStoreWatcher =
+  (electionId: string, store: Store, callback: (electionKey: PublicKey) => void) => () => {
+    const electionKey = getElectionKeyByElectionId(electionId, store.getState());
+
+    if (electionKey) {
+      callback(electionKey);
+    }
+  };
```

Comment:
I don't see this being used anywhere.
Do you really need this ? 
Wouldn't it be enough to update the election object in its reducer and use that in whichever components you need it ?

Reply:
Ah yes, I forgot to remove this! Before the protocol was a bit more complicated and I thought I would need a way to wait for the election#key message being received but this is no longer the case :)